packages/smooth_app/lib/themes/smooth_theme.dart

```diff
@@ -1,63 +1,70 @@
+  APP_BAR_FOREGROUND,
+  APP_BAR_BACKGROUND,
+  SURFACE_FOREGROUND,
+  SURFACE_BACKGROUND,
+  BUTTON_FOREGROUND,
+  BUTTON_BACKGROUND,
+}
+
 class SmoothTheme {
-  static Color getForegroundColor(
-    final ColorScheme colorScheme,
-    final MaterialColor materialColor,
-  ) =>
-      colorScheme.brightness == Brightness.light
-          ? materialColor[800]
-          : materialColor[200];
+  static const double ADDITIONAL_OPACITY_FOR_DARK = .3;
 
-  static Color getBackgroundColor(
+  static Color getColor(
     final ColorScheme colorScheme,
     final MaterialColor materialColor,
-  ) =>
-      colorScheme.brightness == Brightness.light
-          ? materialColor[200]
-          : materialColor[700].withOpacity(.3);
+    final ColorDestination colorDestination,
+  ) {
+    if (colorScheme.brightness == Brightness.light) {
+      switch (colorDestination) {
+        case ColorDestination.APP_BAR_BACKGROUND:
+        case ColorDestination.SURFACE_FOREGROUND:
+        case ColorDestination.BUTTON_BACKGROUND:
+          return materialColor[800];
+        case ColorDestination.APP_BAR_FOREGROUND:
+        case ColorDestination.SURFACE_BACKGROUND:
+        case ColorDestination.BUTTON_FOREGROUND:
+          return materialColor[100];
+      }
+    }
+    switch (colorDestination) {
+      case ColorDestination.APP_BAR_BACKGROUND:
+        return null;
+      case ColorDestination.SURFACE_BACKGROUND:
+      case ColorDestination.BUTTON_BACKGROUND:
+        return materialColor[900].withOpacity(ADDITIONAL_OPACITY_FOR_DARK);
+      case ColorDestination.APP_BAR_FOREGROUND:
+      case ColorDestination.SURFACE_FOREGROUND:
+      case ColorDestination.BUTTON_FOREGROUND:
+        return materialColor[100];
+    }
+    throw Exception(
+      'unknown brightness / destination:'
+      ' ${colorScheme.brightness} / $colorDestination',
+    );
+  }
 
   static ThemeData getThemeData(final Brightness brightness) {
-    final ColorScheme myColorScheme =
-        brightness == Brightness.dark ? _COLOR_DARK : _COLOR_LIGHT;
+    final ColorScheme myColorScheme = brightness == Brightness.dark
+        ? const ColorScheme.dark()
+        : const ColorScheme.light();
     return ThemeData(
       colorScheme: myColorScheme,
       bottomNavigationBarTheme: BottomNavigationBarThemeData(
         selectedItemColor: myColorScheme.onSurface,
         unselectedItemColor: myColorScheme.onSurface,
       ),
       textTheme: _TEXT_THEME,
-      snackBarTheme: SnackBarThemeData(
-        backgroundColor: myColorScheme.onSurface,
-        actionTextColor: myColorScheme.surface,
+      floatingActionButtonTheme: FloatingActionButtonThemeData(
+        backgroundColor: myColorScheme.secondary,
+        foregroundColor: myColorScheme.onSecondary,
       ),
-      appBarTheme: AppBarTheme(
-        color: myColorScheme.background,
-      ),
-      scaffoldBackgroundColor: myColorScheme.background,
     );
   }
 
-  static const ColorScheme _COLOR_LIGHT = ColorScheme.light(
```

Comment:
Just curious: where are the colors of the different themes defined now? e.g. if I want to add a "light blue polar bear" theme, where would I add specify the colors?

Reply:
@stephanegigandet What are used now are the default color schemes for light and dark modes.
```dart
const ColorScheme.light({
  this.primary = const Color(0xff6200ee),
  this.primaryVariant = const Color(0xff3700b3),
  this.secondary = const Color(0xff03dac6),
  this.secondaryVariant = const Color(0xff018786),
  this.surface = Colors.white,
  this.background = Colors.white,
  this.error = const Color(0xffb00020),
  this.onPrimary = Colors.white,
  this.onSecondary = Colors.black,
  this.onSurface = Colors.black,
  this.onBackground = Colors.black,
  this.onError = Colors.white,
  this.brightness = Brightness.light,
})

const ColorScheme.dark({
  this.primary = const Color(0xffbb86fc),
  this.primaryVariant = const Color(0xff3700B3),
  this.secondary = const Color(0xff03dac6),
  this.secondaryVariant = const Color(0xff03dac6),
  this.surface = const Color(0xff121212),
  this.background = const Color(0xff121212),
  this.error = const Color(0xffcf6679),
  this.onPrimary = Colors.black,
  this.onSecondary = Colors.black,
  this.onSurface = Colors.white,
  this.onBackground = Colors.white,
  this.onError = Colors.black,
  this.brightness = Brightness.dark,
})
```

If your question is: is it possible to change the primary color, obviously you'll just have to set `primary` and `primaryVariant` (hopefully not `onPrimary`).
If your question is: how can I _add_ color schemes, like "blue polar bear" and "green deer", for the moment we only have one light and one dark. If we want to deal with several animals and color schemes - like in end-user preferences - it's possible but not managed for the moment.

Reply:
Thanks for the precision. I do want to add a blue polar bear and a green deer at some point indeed. :)